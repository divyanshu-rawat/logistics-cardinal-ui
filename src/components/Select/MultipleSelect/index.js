import React, { Component } from 'react';
import PropTypes from 'prop-types';
import isEqual from 'lodash/isEqual';
import SelectPlus from 'react-select-plus';

import SelectStyled from '../../Select';
import Icons from '../../Icons';
class MultipleSelect extends Component {
  static propTypes = {
    name: PropTypes.string.isRequired,
    options: PropTypes.array.isRequired,
    validationState: PropTypes.oneOf([null, 'success', 'warning', 'error']),
    onChange: PropTypes.func.isRequired,
    size: PropTypes.string,
    placeholder: PropTypes.string,
    isSearchable: PropTypes.bool,
    initialValue: PropTypes.array,
  };

  static defaultProps = {
    initialValue: [],
    size: 'block',
  };

  state = {
    selectedValue: [],
    isOpen: false,
  };

  componentDidMount() {
    const { initialValue } = this.props;

    if (initialValue.length) {
      this.onChange(initialValue);
    }
  }

  componentDidUpdate(nextProps) {
    const { initialValue } = this.props;

    if (!isEqual(nextProps.initialValue, initialValue)) {
      this.onChange(initialValue);
    }
  }

  onChange = (option) => {
    this.setState({ selectedValue: option }, () =>
      this.props.onChange({ option }),
    );
  };

  onOpenSelect = () => {
    this.setState({ isOpen: true });
  };

  onCloseSelect = () => {
    this.setState({ isOpen: false });
  };

  render() {
    const {
      options,
      name,
      placeholder,
      validationState,
      isSearchable,
      size,
    } = this.props;

    const { selectedValue, isOpen } = this.state;

    return (
      <SelectStyled
        size={size}
        isOpen={isOpen}
        validationState={validationState}
      >
        <SelectPlus
          clearable={false}
          searchable={isSearchable}
          joinValues
          name={name}
          multi
          value={selectedValue}
          placeholder={placeholder}
          onOpen={this.onOpenSelect}
          onClose={this.onCloseSelect}
          options={options}
          arrowRenderer={({ isOpen }) =>
            isOpen ? (
              <Icons.ChevronUp className="Custom-Select-arrow" />
            ) : (
              <Icons.ChevronDown className="Custom-Select-arrow" />
            )
          }
          onChange={this.onChange}
        />
      </SelectStyled>
    );
  }
}

export default MultipleSelect;
