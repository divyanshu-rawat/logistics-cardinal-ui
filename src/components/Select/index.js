import styled from 'styled-components';

const sizes = {
  small: '100px',
  regular: '300px',
  block: '100%',
};

const validationStateColor = {
  error: 'inputErrorBorderColor',
  success: 'successColor',
  warning: 'warningColor',
};

const SelectStyled = styled('div')`
  display: inline-block;
  margin-right: 5px;
  width: 100%;
  position: relative;
  min-width: ${sizes.small};
  z-index: ${({ theme, isOpen }) =>
    isOpen ? theme.zIndex.select + 1 : theme.zIndex.select};
  .Select {
    width: ${({ size }) => sizes[size]};
    &-control {
      width: ${({ size }) => sizes[size]};
      background-color: ${({ theme }) => theme.inputBackgroundColor};
      border: 1px solid
        ${({ validationState, theme }) =>
          validationState
            ? theme[validationStateColor[validationState]]
            : theme.inputBorderColor};
      display: table;
      border-spacing: 0;
      border-collapse: separate;
      align-items: center;
      height: ${({ theme }) => theme.inputDefaultHeight};
      &:hover {
        cursor: pointer;
        border-color: ${({ theme }) => theme.inputBorderColorHover};
      }
    }
    &.is-focused,
    &.is-open {
      > .Select-control {
        background-color: ${({ theme }) => theme.inputBackgroundColor};
        border: 1px solid
          ${({ validationState, theme }) =>
            validationState
              ? theme[validationStateColor[validationState]]
              : theme.inputBorderColor};
        box-shadow: none;
      }
    }
    &.is-disabled {
      > .Select-control {
        background-color: ${({ theme }) => theme.inputDisabledBackgroundColor};
        &:hover {
          cursor: default;
          border: 1px solid ${({ theme }) => theme.inputBorderColor};
        }
      }
    }
    &.is-searchable {
      &.is-open > .Select-control {
        cursor: text;
      }

      &.is-focused:not(.is-open) > .Select-control {
        cursor: text;
      }
    }
    &-multi {
      &-value-wrapper {
        flex: 1;
      }
    }
    &-placeholder {
      line-height: ${({ theme }) => theme.inputDefaultHeight};
      color: ${({ theme }) => theme.inputPlaceholderColor};
    }
    &-value {
      margin: 0 0 0 5px;
    }
    &-input {
      background: transparent;
      color: ${({ theme }) => theme.textColor};
      height: 38px;
      > input {
        color: inherit;
        padding: 13px 0 9px;
      }
      &:focus {
        background: transparent;
      }
    }
    &-arrow {
      background-color: ${({ theme }) => theme.primaryColor};
    }
    &-menu-outer {
      box-shadow: 0;
      background-color: ${({ theme }) => theme.inputBackgroundColor};
      border: 1px solid ${({ theme }) => theme.inputBorderColor};
      .Select-menu .Select-option {
        background-color: ${({ theme }) => theme.inputBackgroundColor};
        color: ${({ theme }) => theme.textColor};
        &.is-selected {
          background-color: ${({ theme }) => theme.selectItemFocus};
          border-bottom: 1px solid ${({ theme }) => theme.inputBorderColor};
        }
        &.is-focused {
          background-color: ${({ theme }) => theme.selectItemFocus};
          border: 0;
        }
        &:hover {
          background-color: ${({ theme }) => theme.selectItemFocus};
          border: 0;
        }
        &.is-selected:hover {
          background-color: ${({ theme }) => theme.selectItemFocus};
          border-bottom: 1px solid ${({ theme }) => theme.inputBorderColor};
        }
        &.is-disabled {
          color: ${({ theme }) => theme.selectItemDisabledColor};
          &:hover {
            background-color: ${({ theme }) =>
              theme.selectItemDisabledHoverBackgroundColor};
            color: ${({ theme }) => theme.selectItemDisabledColor};
          }
        }
      }
    }
    &--multi {
      .Select-value {
        display: inline-flex;
        background-color: ${({ theme }) => theme.selectMultiTagBackgroundColor};
        color: ${({ theme }) => theme.selectMultiTagTextColor};
        flex-flow: row-reverse;
        align-items: center;
        &-label {
          flex: 1;
        }
      }
      .Select-multi-value-wrapper {
        .Select-value {
          margin-top: 9px;
        }
      }
    }
    &--single {
      > .Select-control .Select-value {
        margin: 0;
        bottom: 0;
        left: 0;
        padding-left: 10px;
        padding-right: 10px;
        position: absolute;
        right: 0;
        top: 0;
        max-width: 100%;
        overflow: hidden;
        text-overflow: ellipsis;
        line-height: ${({ theme }) => theme.inputDefaultHeight};
        white-space: nowrap;
      }
    }
    &.has-value {
      &.Select--single {
        > .Select-control .Select-value .Select-value-label {
          color: ${({ theme }) => theme.selectSingleTextColor};
        }
      }
      &.is-pseudo-focused .Select-input {
        opacity: 0;
      }
    }
  }
`;

export default SelectStyled;
