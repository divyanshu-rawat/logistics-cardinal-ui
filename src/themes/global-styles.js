import { createGlobalStyle } from 'styled-components';
import 'normalize.css';

export const GlobalStyles = createGlobalStyle`
  ${({ theme, injectGlobal }) => `
    *,
    *::before,
    *::after {
      box-sizing: border-box;
    }

    html {
      box-sizing: border-box;
      line-height: 1.15;
      -webkit-text-size-adjust: 100%;
      -ms-text-size-adjust: 100%;
      -ms-overflow-style: scrollbar;
      -webkit-tap-highlight-color: rgba(0, 0, 0, 0);
      [type='button'] {
        appearance: none;
      }
    }

    body {
      margin: 0;
      background-color: ${theme.bodyBackground};
      color: ${theme.textColor};
      font-family: ${theme.textFontFamily};
      font-size: ${theme.textFontSize};
      font-weight: ${theme.textFontWeight};
      line-height: ${theme.lineHeightBase};
    }

    fieldset {
      min-width: 0;
      padding: 0;
      margin: 0;
      border: 0;
    }

    .Drawer {
      &--is-open {
        overflow: hidden;
      }
    }

    .ReactModal {
      &__Body--open {
        overflow: hidden;
      }
      &__Overlay {
        position: fixed;
        top: 0;
        left: 0;
        right: 0;
        bottom: 0;
        transition: opacity 200ms ease-in-out;
        height: auto;
        overflow: auto;
        z-index: ${theme.zIndex.modalOverlay};
        background-color: ${theme.overlayColor};
        &--after-open {
          opacity: 1;
        }
        &--before-close {
          opacity: 0;
        }
      }
      &__Content {
        margin: 10em auto;
        position: relative;
        left: auto;
        right: auto;
        top: auto;
        bottom: auto;
        transform: none;
        opacity: 0;
        transition: opacity 200ms ease-in-out;
        min-height: ${theme.modalMinHeight};
        min-width: ${theme.modalMinWidth};
        width: ${theme.modalWidth};
        max-width: ${theme.modalMaxWidth};
        background-color: ${theme.modalBackgroundColor};
        border: 1px ${theme.modalBorderStyle} ${theme.modalBorderColor};
        box-shadow: ${theme.modalBoxShadow};
        border-radius: ${theme.modalBorderRadius};
        outline: ${theme.modalOutline};
        z-index: ${theme.zIndex.modalContent};
        &--after-open {
          opacity: 1;
        }
        &--before-close {
          opacity: 0;
        }
      }
    }


    ${injectGlobal && injectGlobal(theme)};
  `}
`;
