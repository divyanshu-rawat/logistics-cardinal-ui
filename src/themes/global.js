import { injectGlobal } from 'styled-components';

import 'react-select/dist/react-select.css';
import 'react-select-plus/dist/react-select-plus.css';

/* eslint no-unused-expressions: 0 */
injectGlobal`
  @import url('https://fonts.googleapis.com/css?family=Open+Sans:300,400,600,700,800');

  // used to overwrite bootstrap's background-color
  table[class^="css-"] {
    background: white;
    td[class^="css-"] {
      white-space: initial;
    }
  }
`;
