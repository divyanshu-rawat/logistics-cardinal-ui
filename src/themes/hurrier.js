const white = '#fff';
const black = '#000';

const primary = {
  primary100: '#db0062',
  primary200: '#375f8a',
};

const greys = {
  grey100: '#96a2b0',
  grey200: '#3f454d',
  grey300: '#343a42',
  grey400: '#292d33',
};

const greens = {
  green100: '#c6e5cb',
  green300: '#97e2a3',
  green500: '#62de76',
  green700: '#49b85b',
  green900: '#2f903e',
};

const reds = {
  red100: '#ffe6e1',
  red300: '#ffaf9f',
  red500: '#ff7559',
  red700: '#d55a41',
  red900: '#a73d28',
};

const yellows = {
  yellow100: '#f2e5cb',
  yellow300: '#f7da9f',
  yellow500: '#ffc859',
  yellow700: '#d4a546',
  yellow900: '#ab8433',
};

const misc = {
  bodyBackground: '#fff',
  mainBackground: '#ccc',
  textColor: white,
  danger: reds.red500,
  success: greens.green700,
  warning: yellows.yellow700,
};

const colors = {
  white,
  black,
  ...primary,
  ...greys,
  ...greens,
  ...reds,
  ...yellows,
  ...misc,
};

const spacings = {
  bit: '4px',
  byte: '8px',
  kilo: '12px',
  mega: '16px',
  giga: '24px',
  tera: '32px',
  peta: '40px',
  exa: '48px',
  zetta: '56px',
};

const borderRadius = {
  kilo: '1px',
  mega: '4px',
  giga: '5px',
};

const borderWidth = {
  kilo: '1px',
  mega: '2px',
};

const fontFamily = {
  default: '"Open Sans", sans-serif',
};

const fontWeight = {
  regular: '400',
  bold: '700',
};

export default {
  colors,
  spacings,
  borderRadius,
  borderWidth,
  fontFamily,
  fontWeight,
};
