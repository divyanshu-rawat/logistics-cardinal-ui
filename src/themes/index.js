export rooster from './rooster';
export standard from './standard';
export hurrier from './hurrier';
