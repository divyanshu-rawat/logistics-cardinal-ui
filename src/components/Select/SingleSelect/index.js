import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import SelectPlus from 'react-select-plus';

import Select from '../../Select';

class SingleSelect extends PureComponent {
  static propTypes = {
    options: PropTypes.array.isRequired,
    onChange: PropTypes.func.isRequired,
    name: PropTypes.string.isRequired,
    placeholder: PropTypes.string.isRequired,
    isSearchable: PropTypes.bool,
    isFirst: PropTypes.bool,
    size: PropTypes.string,
    initialValue: PropTypes.string,
  };

  state = {
    selectedValue: this.props.initialValue,
  };

  onChangeTime = (option) => {
    const { name } = this.props;
    this.setState({ selectedValue: option }, () =>
      this.props.onChange({ name, option }),
    );
  };

  render() {
    const {
      size = 'block',
      options,
      name,
      placeholder,
      isSearchable = false,
      isFirst,
    } = this.props;

    const { selectedValue } = this.state;

    return (
      <Select size={size} isFirst={isFirst}>
        <SelectPlus
          joinValues
          clearable={false}
          searchable={isSearchable}
          name={name}
          value={selectedValue}
          placeholder={placeholder}
          options={options}
          onChange={this.onChangeTime}
        />
      </Select>
    );
  }
}

export default SingleSelect;
