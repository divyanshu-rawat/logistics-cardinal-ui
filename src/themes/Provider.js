import React, { Fragment } from 'react';
import { ThemeProvider } from 'styled-components';
import PropTypes from 'prop-types';

import { standard } from '.';
import { GlobalStyles } from './global-styles';

const CardinalThemeProvider = ({ theme, injectGlobal, children }) => {
  const mergedTheme = {
    ...standard,
    ...theme,
  };

  return (
    <ThemeProvider theme={mergedTheme}>
      <Fragment>
        <GlobalStyles injectGlobal={injectGlobal} />
        {children}
      </Fragment>
    </ThemeProvider>
  );
};

CardinalThemeProvider.propTypes = {
  children: PropTypes.node.isRequired,
  injectGlobal: PropTypes.func,
  theme: PropTypes.object,
};

export { CardinalThemeProvider };
