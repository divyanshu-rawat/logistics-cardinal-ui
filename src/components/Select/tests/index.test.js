import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { rooster as RoosterTheme } from '../../../themes';

import Select from '..';

const renderedComponent = ({ size = '', validationState = null }) =>
  shallow(
    <Select
      theme={RoosterTheme}
      size={size}
      validationState={validationState}
    />,
  );

describe('<Select />', () => {
  it('should render the component with `default size`', () => {
    const tree = toJson(renderedComponent({}));
    expect(tree).toMatchSnapshot();
  });

  it('should render the component with `small size`', () => {
    const tree = toJson(renderedComponent({ size: 'small' }));
    expect(tree).toMatchSnapshot();
  });

  it('should render the component with `error validation state`', () => {
    const tree = toJson(renderedComponent({ validationState: 'error' }));
    expect(tree).toMatchSnapshot();
  });

  it('should render the component with `warning validation state`', () => {
    const tree = toJson(renderedComponent({ validationState: 'warning' }));
    expect(tree).toMatchSnapshot();
  });

  it('should render the component with `success validation state`', () => {
    const tree = toJson(renderedComponent({ validationState: 'success' }));
    expect(tree).toMatchSnapshot();
  });
});
