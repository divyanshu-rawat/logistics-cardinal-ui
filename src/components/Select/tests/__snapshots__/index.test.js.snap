// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Select /> should render the component with \`default size\` 1`] = `
<div
  className="Select__SelectStyled-fwaYyW eoLBzF"
  size=""
/>
`;

exports[`<Select /> should render the component with \`error validation state\` 1`] = `
<div
  className="Select__SelectStyled-fwaYyW eoFsJg"
  size=""
/>
`;

exports[`<Select /> should render the component with \`small size\` 1`] = `
<div
  className="Select__SelectStyled-fwaYyW foxtNb"
  size="small"
/>
`;

exports[`<Select /> should render the component with \`success validation state\` 1`] = `
<div
  className="Select__SelectStyled-fwaYyW ijUDPn"
  size=""
/>
`;

exports[`<Select /> should render the component with \`warning validation state\` 1`] = `
<div
  className="Select__SelectStyled-fwaYyW cwTJME"
  size=""
/>
`;
